go-controller/pkg/ovn/ovn.go

```diff
@@ -324,20 +324,38 @@ func (oc *Controller) ovnControllerEventChecker(stopChan chan struct{}) {
 	}
 }
 
+func podNeedsSetup(pod *kapi.Pod) bool {
+	// Unscheduled pods don't need logical ports
+	if pod.Spec.NodeName == "" {
+		return false
+	}
+
+	// Pods with an annotation were already handled
+	if _, err := util.UnmarshalPodAnnotation(pod.Annotations["ovn"]); err == nil {
+		return false
+	}
+
+	// Scheduled but unhandled; need a logical port
+	return true
+}
+
 // WatchPods starts the watching of Pod resource and calls back the appropriate handler logic
 func (oc *Controller) WatchPods() error {
 	_, err := oc.watchFactory.AddPodHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			pod := obj.(*kapi.Pod)
+			// Always set up scheduled pods even if they have an
+			// existing annotation; this ensures the OVN database is
+			// synced/updated on restart
```

Comment:
but if `addLogicalPort` fails, we won't retry on Update...

Reply:
Fixed.